Swabbr.Infrastructure/Extensions/SwabbrInfrastructureServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using Swabbr.Core.Interfaces.Repositories;
+using Swabbr.Core.Interfaces.Services;
+using Swabbr.Core.Notifications;
+using Swabbr.Infrastructure.Database;
+using Swabbr.Infrastructure.Notifications;
+using Swabbr.Infrastructure.Notifications.JsonExtraction;
+using Swabbr.Infrastructure.Providers;
+using Swabbr.Infrastructure.Repositories;
+using System;
+
+namespace Swabbr.Infrastructure.Extensions
+{
+    /// <summary>
+    ///     Extends the service collection for the Swabbr 
+    ///     infrastructure package.
+    /// </summary>
+    public static class SwabbrInfrastructureServiceCollectionExtensions
+    {
+        /// <summary>
+        ///     Adds the Swabbr Infrastructure services to the
+        ///     service collection.
+        /// </summary>
+        /// <param name="services">The service collection.</param>
+        /// <returns>Same as <paramref name="services"/>.</returns>
+        public static IServiceCollection AddSwabbrInfrastructureServices(this IServiceCollection services, string dbConnectionStringName)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            // Add postgresql database functionality
+            NpgsqlSetup.Setup();
```

Comment:
This restricts the class from chaning impl.

Reply:
100% dapper specific. Will be removed in future, marked as TODO.